src/mats_l1_processing/read_parquet_functions.py

```diff
@@ -0,0 +1,198 @@
+
+    if "WDW InputDataWindow" not in item:
+        item["WDW InputDataWindow"] = item.pop("WDWInputDataWindow")
+
+    if "NCBIN CCDColumns" not in item:
+        item["NCBIN CCDColumns"] = item.pop("NCBINCCDColumns")
+
+    if "NCBIN FPGAColumns" not in item:
+        item["NCBIN FPGAColumns"] = item.pop("NCBINFPGAColumns")
+
+    if "GAIN Mode" not in item:
+        item["GAIN Mode"] = item.pop("GAINMode")
+
+    if "GAIN Timing" not in item:
+        item["GAIN Timing"] = item.pop("GAINTiming")
+
+    if "GAIN Truncation" not in item:
+        item["GAIN Truncation"] = item.pop("GAINTruncation")
+
+    if "BC" not in item:
+        item["BC"] = item.pop("BadColumns")
+    
+    if "Image File Name" not in item:
+        item["Image File Name"] = item.pop("ImageName")
+
+    if "IMAGE" not in item:
+        item["IMAGE"] = np.float64(Image.open(BytesIO(item.pop["ImageData"])))
+
+
+def add_CCDItem_attributes(item: CCDItem):
+    """Add some attributes to CCD Item that we need. The names in the code
+    are based on the old rac extract file (prior to May 2020).
+
+    Args:
+        item: Item from Parquet store to which to add attributes.
+    
+    Returns:
+        Nothing. Operation is performed in place.
+    """
+
+    item["channel"] = channel_num_to_str[item["CCDSEL"]]
+    item["flipped"] = False
+
+    # CCDitem["id"] should not be needed in operational retrieval. Keeping it
+    # because protocol reading / CodeCalibrationReport needs it.  LM220908
+    item["id"] = f"{item['EXP Nanoseconds']}_{item['CCDSEL']}"
+
+    # Add temperature info fom OBC, the temperature info from the rac files are
+    # better since they are based on the thermistors on the UV channels
+    ADC_temp_in_mV = int(item["TEMP"]) / 32768 * 2048
+    ADC_temp_in_degreeC = 1.0 / 0.85 * ADC_temp_in_mV - 296
+    item["temperature_ADC"] = ADC_temp_in_degreeC
+
+    # TODO: Replace with something like `add_temperature_info` when HTR data is
+    # available, see: https://github.com/innosat-mats/level1a/issues/5
+    item["temperature"] = ADC_temp_in_degreeC
+    item["temperature_HTR"] = ADC_temp_in_degreeC
+
+
+def add_and_rename_CCDitem_attributes(items: List[CCDItem]):
+    """Applies renaming of CCD attributes as well as adding a few that are
+    needed. Note that items with errors are removed, so the list may be shorter
+    after processing.
+
+    Args:
+        items: List of items from Parquet store to be converted to CCCD items.
+    
+    Returns:
+        Nothing. Operation is performed in place.
+    """
+
+    for ind, item in enumerate(items):
+        try:
+            rename_CCDitem_attributes(item)
+            add_CCDItem_attributes(item)
+        except Exception as err:
+            print(
+                f"Warning: could not prepare image {item.get('ImageName', 'unknown')} for calibration. Skipping. (Error: {err})"  # noqa: E501
+            )
+            items.pop(ind)
+
+
+def read_CCDitems_interval(
```

Comment:
Do you think there is a possibility that the processing functions will be rewritten to work on `DataFrames` rather than `dicts`? If so, consider splitting this and the function below to return a `DataFrame` and then add something like `dataframe_to_CCDitems()`. Might be very yagni tho...

Reply:
Yes, see #70 ;) I'll see if I can split it already today.

Reply:
I have refactored to use `DataFrame` until the final return, which should make transitioning to #70 easier.

This created a few functions that are ridiculously short compared to their docstrings, but maybe that's all right?

If you wouldn't mind, could you review again?